### PR TITLE
Verify block slot

### LINKED ISSUE #1998: Add BlockSlot validation
https://github.com/iotaledger/tips/commit/688788a5ef5e711b103042070a6c02a6aaba297a

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #1998

### CODE REVIEW

File: sdk/src/types/block/core/block.rs [hunk 1]
```diff
@@ -154,17 +173,18 @@ impl Block {
...
 
-        Self {
-            header,
-            body,
-            signature,
-        }
-    }
+    // }
+
+    // TODO add new_unverified
```
Comment (Thoralf-M) [inline sdk/src/types/block/core/block.rs]: Want to do in this PR or another one?

File: sdk/src/types/block/error.rs [hunk 2]
```diff
@@ -450,6 +453,16 @@ impl fmt::Display for Error {
             }
             Self::TrailingCapabilityBytes => write!(f, "capability bytes have trailing zeroes"),
             Self::RestrictedAddressCapability(cap) => write!(f, "restricted address capability: {cap:?}"),
+            Self::BlockSlotBeforeTransactionCreationSlot => {
+                write!(f, "the block slot is before its contained transaction creation slot")
+            }
+            Self::TransactionCommitmentSlotNotInBlockSlotInterval => write!(
+                f,
+                "the transaction commitment slot int not in the allowed block slot interval"
```
Comment (Thoralf-M) [inline sdk/src/types/block/error.rs]:
```suggestion
                "the transaction commitment slot is not in the allowed block slot interval"
```

File: sdk/src/types/block/core/block.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/core/block.rs]: No actually I forgot to remove the comment, we decided with Chloé to do with `UnsignedBlock::finish` completely and remove `Block::new`

Comment (thibault-martinez) [reply sdk/src/types/block/core/block.rs]: Done

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/types/block/core/block.rs [hunk 3]
```diff
@@ -304,6 +313,35 @@ impl Packable for Block {
     }
 }
 
+fn verify_block_slot(header: &BlockHeader, body: &BlockBody, params: &ProtocolParameters) -> Result<(), Error> {
+    if let BlockBody::Basic(basic) = body {
+        if let Some(Payload::SignedTransaction(signed_transaction)) = basic.payload() {
+            let transaction = signed_transaction.transaction();
+            let block_slot = params.slot_index(header.issuing_time / 1_000_000_000);
```
Comment (Alex6323) [inline sdk/src/types/block/core/block.rs:320]: Is this calculation worth a utils method? this is basically `to_sec`?

**Review by Alex6323 — APPROVED** [review]

File: sdk/src/types/block/core/block.rs (hunk 3 above)
Comment (thibault-martinez) [reply sdk/src/types/block/core/block.rs:320]: I don't think so, we can add when/if there is other occurrences of this

**Pull request merged**